model/registry/process.yaml

```diff
@@ -11,6 +11,31 @@ groups:
         brief: >
           Process identifier (PID).
         examples: [1234]
+      - id: entity_id
+        type: string
+        stability: experimental
+        brief: >
+          A unique identifier of the process.
+        note: |
+          A globally unique identifier of the process. This can be any format
+          of unique indentifier.
+
+          On Linux, where possible, it is RECOMMENDED that this is calculated
```

Comment:
do we need to add any recommendations for other OS?

Reply:
If anyone knows of a way to generate unique deterministic entity_ids for other OSs, we can definitely add it.

I thought about other OSs, but I couldn't really find a way to do it for OSs that don't have a boot UUID.

These are some of the things I thought about though:

* pid + machine/hardware ID + time_since_boot is not usable, because it could collide on reboots.
* Wall clock is not usable, because it can move backwards and cause collisions.
* I don't think any standard UUID version is usable, for various reasons, but basically they are designed not to collide, and we will want them to collide and generate the same UUID for the process.
* There could be some inter-process communication to share IDs between collectors, but that might be difficult to have collectors agree to use it and seems like it's outside of the scope of semantic conventions